### PR TITLE
use neovim treesitter API, remove dependency on nvim-treesitter

### LINKED ISSUE #63: Feature request: support nvim-treesitter next version
See:

- https://github.com/nvim-treesitter/nvim-treesitter/tree/main
- https://github.com/nvim-treesitter/nvim-treesitter/issues/4767

### PR DESCRIPTION
**Pull request by mrcjkb**
Closes #63.

### CODE REVIEW

File: lua/tabout/node.lua [hunk 1]
```diff
@@ -33,16 +31,21 @@ end
...
+    if parser then
+      parser:parse()
+    else
+        logger.debug("get_node_at_cursor: No parser found for filetype " .. vim.bo[0].filetype)
+        return
     end
-    local root =
-        ts_utils.get_root_for_position(cursor_range[1], cursor_range[2])
+    local tree = parser:tree_for_range(cursor_range)
+    local root = tree and tree:root()
```
Comment (abecodes) [inline lua/tabout/node.lua:49]: Some changes here cause line 62 to throw an error for me on a simple go file:

```
//test.go
package main

func main(|) // hit tab here
```

Error about concatenating a bool value in line 62.

Can you double check or let me know what I am missing?

**Review by abecodes — CHANGES_REQUESTED** [review]

**Comment by abecodes** [comment]
Thanks for the submission, really appreciated 🎖️ 👍

File: lua/tabout/node.lua (hunk 1 above)
Comment (mrcjkb) [reply lua/tabout/node.lua:49]: Oops. Fixed :smile:

**Review by abecodes — APPROVED** [review]

**Comment by abecodes** [comment]
Thanks for the submission and the fix @mrcjkb 👍 🎖️ 
Really appreciate your hard work 🙇

**Pull request merged**